### PR TITLE
change: add buildspec-release file and upgrade cuda version

### PR DESCRIPTION
**Pull request by ChuyangDeng**
*Issue #, if available:*

*Description of changes:*
- add buildspec-release.yml
- add cpu_py3 and gpu_py2 test run in buildspec.yml
- fix tox coverage

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=257a0ec5-2501-489d-952d-38fface06871%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=1b467835-2e98-451b-9be1-f30aa7f194fd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ccbbffd9-cac6-46a1-8cc4-7a3683a27a85%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=66dd9671-8411-47f0-a352-613cadea9a56%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=4568f50c-3c28-4ebc-bc8a-0e90d8c3ce64%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=5a5bd384-ae66-4061-8493-68c3247b01f3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=6ebcba6c-b5d5-43ad-9576-f694a26b27bf%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=8c8d8755-e8ce-4ae4-acf4-bc4395856585%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c19ec671-cb0e-4e7c-977a-51378a9fbdae%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ac14d537-83fd-4cdb-84c7-3b6f291b7a54%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=826afccc-c858-4a21-a513-bfae95e448cd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=e7ce62a2-e319-4638-ab72-b065e515821d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=e62a0075-3e7e-4544-bb83-46b4acae8414%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=9624e951-13f3-477b-8aeb-eeb45f8d731a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=318c5b4a-4f80-4576-9548-b1bd0b3ad94f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=afaad10e-e671-40dc-b490-a4a192841702%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=6d276867-f10e-4f7b-965e-6f820ec0d369%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=2116f2b5-0e86-4c9f-94e3-a5179f43b5a4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://vvaldn0il2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=089f6c6c-e090-4fa6-9098-e22b8bc5e4b5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: test/integration/sagemaker/test_distributed_operations.py [hunk 1]
```diff
@@ -44,11 +44,8 @@ def test_dist_operations_multi_gpu(sagemaker_session, ecr_image, dist_gpu_backen
 
 
 @pytest.mark.skip_cpu
-def test_dist_operations_fastai_gpu(sagemaker_session, ecr_image, py_version):
-    if py_version != PYTHON3:
-        print('Skipping the test because fastai supports >= Python 3.6.')
-        return
-
+@pytest.mark.skip_py2
```
Comment (mvsusp) [inline test/integration/sagemaker/test_distributed_operations.py]: You should skip both CPU and PY2 here.

File: buildspec-release.yml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+      - echo 'Pull request number:' $PR_NUM '. No value means this build is not from pull request.'
+
+  build:
+    commands:
+      # prepare the release
+      - git-release --prepare --min-version 1.2.0
+
+      # install
+      - pip3 install -U -e .
+      - pip3 install -U -e .[test]
```
Comment (mvsusp) [inline buildspec-release.yml]: Do you need to install these dependencies here with pip3?

Comment (ChuyangDeng) [reply buildspec-release.yml]: I was following the buildspec.yml using pip3, but I'll ask Yang about it.

Comment (ChuyangDeng) [reply buildspec-release.yml]: Both python 2 and 3 versions of pip are installed in the new build system, so we need to specify pip3.

File: test/integration/sagemaker/test_distributed_operations.py (hunk 1 above)
Comment (ChuyangDeng) [reply test/integration/sagemaker/test_distributed_operations.py]: This test already skips cpu on the line above

**Pull request merged**